insights/v1alpha1/types_insights.go

```diff
@@ -0,0 +1,198 @@
+	// +optional
+	DisabledGatherers []string `json:"disabledGatherers"`
+}
+
+const (
+	// No data obfuscation
+	NoPolicy DataPolicy = "None"
+	// IP addresses and cluster domain name are obfuscated
+	ObfuscateNetworking DataPolicy = "ObfuscateNetworking"
+	// Data gathering is running
+	Running State = "Running"
+	// Data gathering is completed
+	Completed State = "Completed"
+	// Data gathering failed
+	Failed State = "Failed"
+	// Data gathering is pending
+	Pending State = "Pending"
+)
+
+// dataPolicy declares valid data policy types
+// +kubebuilder:validation:Enum="";None;ObfuscateNetworking
+type DataPolicy string
+
+// state declares valid gathering state types
+// +kubebuilder:validation:Enum=Running;Completed;Failed
+type State string
+
+type DataGatherStatus struct {
+	// conditions provide details on the status of the gatherer job.
+	// +listType=atomic
+	// +optional
+	Conditions []metav1.Condition `json:"conditions"`
+	// state reflects the current state of the data gathering process.
+	// +optional
+	State State `json:"state,omitempty"`
+	// gatherers is a list of active gatherers (and their statuses) in the last gathering.
+	// +listType=atomic
+	// +optional
+	Gatherers []GathererStatus `json:"gatherers,omitempty"`
+	// startTime is the time when Insights data gathering started.
+	// +optional
+	StartTime metav1.Time `json:"startTime,omitempty"`
+	// finishTime is the time when Insights data gathering finished.
+	// +optional
+	FinishTime metav1.Time `json:"finishTime,omitempty"`
+	// relatedObjects is a list of resources which are useful when debugging or inspecting the data
+	// gathering Pod
+	// +optional
+	RelatedObjects []ObjectReference `json:"relatedObjects,omitempty"`
+	// insightsReport provides general Insights analysis results.
+	// When omitted, this means no data gathering has taken place yet.
+	// +optional
+	InsightsReport InsightsReport `json:"insightsReport,omitempty"`
+}
+
+// gathererStatus represents information about a particular
+// data gatherer.
+type GathererStatus struct {
+	// conditions provide details on the status of each gatherer.
+	// +listType=atomic
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:MinItems=1
+	Conditions []metav1.Condition `json:"conditions"`
+	// name is the name of the gatherer.
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:MaxLength=256
+	// +kubebuilder:validation:MinLength=5
+	Name string `json:"name"`
+	// lastGatherDuration represents the time spent gathering.
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:Type=string
+	// +kubebuilder:validation:Pattern="^([1-9][0-9]*(\\.[0-9]+)?(ns|us|µs|ms|s|m|h))+$"
+	LastGatherDuration metav1.Duration `json:"lastGatherDuration"`
+}
+
+// insightsReport provides Insights health check report based on the most
+// recently sent Insights data.
+type InsightsReport struct {
```

Comment:
Do we want to provide a link to the original source for the download? I think there were some comments like that in the enhancement?

Reply:
Interesting question. I wondered why at first, but I guess there's no harm in adding it. Most of the time it will be the same, but it might be useful for observability (general information).